op-proposer/proposer/driver.go

```diff
@@ -269,18 +269,48 @@ func (l *L2OutputSubmitter) FetchL2OOOutput(ctx context.Context) (*eth.OutputRes
 	return output, true, nil
 }
 
-// FetchDGFOutput gets the next output proposal for the DGF.
+// FetchDGFOutput queries the DGF for the latest game and infers whether it is time to make another proposal
+// If necessary, it gets the next output proposal for the DGF, and returns it along with
+// a boolean for whether the proposal should be submitted at all.
 // The passed context is expected to be a lifecycle context. A network timeout
 // context will be derived from it.
-func (l *L2OutputSubmitter) FetchDGFOutput(ctx context.Context) (*eth.OutputResponse, error) {
+func (l *L2OutputSubmitter) FetchDGFOutput(ctx context.Context) (*eth.OutputResponse, bool, error) {
 	ctx, cancel := context.WithTimeout(ctx, l.Cfg.NetworkTimeout)
 	defer cancel()
 
-	blockNum, err := l.FetchCurrentBlockNumber(ctx)
+	shouldPropose := false
+
+	gameCount, err := l.dgfContract.GameCount(&bind.CallOpts{Context: ctx})
 	if err != nil {
-		return nil, err
+		l.Log.Warn("Could not query DisputeGameFactory.gameCount(), sending a proposal immediately")
+		shouldPropose = true
+	} else {
+		latestGameIndex := new(big.Int).Sub(gameCount, big.NewInt(1))
+		latestGame, err := l.dgfContract.GameAtIndex(&bind.CallOpts{Context: ctx}, latestGameIndex)
+		if err != nil {
+			l.Log.Warn(fmt.Sprintf(
+				"Could not query DisputeGameFactory.GameAtIndex(%d), sending a proposal immediately",
+				latestGameIndex.Int64()))
+			shouldPropose = true
+		} else {
+			timeSinceLastGame := time.Since(time.Unix(int64(latestGame.Timestamp), 0))
+			if timeSinceLastGame > l.Cfg.ProposalInterval {
+				shouldPropose = true
+				l.Log.Info(fmt.Sprintf("More than %f seconds since last game, sending a proposal now", timeSinceLastGame.Seconds()))
```

Comment:
It would probably be nicer to read the Go duration formatting, which would be in the form `1h2m24s` by using `%v` and just `timeSinceLastGame `, or just use structured logging:
```suggestion
				l.Log.Info("Duration since last game past proposal interval", "duration", timeSinceLastGame)
```

Reply:
[e9f2658](https://github.com/ethereum-optimism/optimism/pull/11417/commits/e9f2658eb8eb1c00a463dd3f6f2e98eec54bba83)